src/web/page/web_page.rs

```diff
@@ -0,0 +1,57 @@
+use super::TEMPLATE_DATA;
+use iron::{headers::ContentType, response::Response, status::Status, IronResult};
+use serde::Serialize;
+use tera::Context;
+
+#[macro_export]
+macro_rules! impl_webpage {
+    ($page:ty = $template:expr $(, status = $status:expr)? $(, content_type = $content_type:expr)? $(,)?) => {
+        impl $crate::web::page::WebPage for $page {
+            const TEMPLATE: &'static str = $template;
+
+            $(
+                fn get_status(&self) -> ::iron::status::Status {
```

Comment:
`&self` doesn't look necessary here.

Reply:
It is, the status can be gotten from the template itself

Reply:
What do you mean? When is this used?

Reply:
For error pages

Reply:
I'm still not quite following but we can revisit this when you add the error pages and it's more clear why it's necessary.